File: src/project/src/rocket_controller.cpp
#include <ros/ros.h>
#include <lab2/angle_msg.h>
#include <lab3/ballLandInfo.h>
#include <fanboat_ll/fanboatMotors.h>
#include <fanboat_ll/fanboatLL.h>
#include <math.h>
#include <project/rocket_msg.h>
using namespace std;

class rocket_controller{
public:
  rocket_controller();
private:
  void msgCallback(const lab3::ballLandInfo::ConstPtr& msg);

  ros::NodeHandle nh;

  //Subscriber for landmark
  ros::Subscriber img_arb_sub_;

  //Publisher for angle
  ros::Publisher angle_pub_;
  ros::Publisher rocket_pub_;
};

rocket_controller::rocket_controller()
{
  img_arb_sub_ = nh.subscribe("ballLandInfo", 1, &rocket_controller::msgCallback, this);

  //Landmark position publisher
  angle_pub_ = nh.advertise<lab2::angle_msg>("angle_cam",1);

  rocket_pub_ = nh.advertise<project::rocket_msg>("rocket_command",1);
}

void rocket_controller::msgCallback(const lab3::ballLandInfo::ConstPtr& msg)
{
	int x = msg->x;
	int y = msg->y;
	float distance = msg->distance;

	int diffx = 300 - x;
  double diffang = (diffx/300.0) * 22.5;

	project::rocket_msg rocket;
	rocket.angle = diffang;
	//ang.thrust = 0.3;
  rocket_pub_.publish(rocket);
}


int main(int argc, char** argv)
{
  ros::init(argc, argv, "rocket_controller");
  rocket_controller rc;
  ros::spin();
  return 0;
}

File: src/project/src/rocket_controller.py
#!/usr/bin/env python

import rospy
from project.msg import rocket_msg
from lab3.msg import ballLandInfo
from lab2.msg import angle_msg
from fanboat_ll.msg import fanboatMotors
from fanboat_ll.msg import fanboatLL

class rocket_controller(object):
    rocketPub = None
    anglePub = None
    #subscriber1 = None
    #subscriber2 = None

    #launcher = armageddon.Armageddon()
    def __init__(self):
        rospy.init_node('rocket_controller')

    def init_subcribers(self):
        print "init subs"
        #rospy.Subscriber('/rocket_command', rocket_msg, self.rocket_callback)
        rospy.Subscriber('/ballLandInfo', ballLandInfo, self.camCallback)

    def init_publishers(self):
        print "init pubs"
        self.rocketPub = rospy.Publisher('rocket_topic', rocket_msg, queue_size=1)
        self.anglePub = rospy.Publisher('angle_topic', angle_msg, queue_size=1)

    def calculateTurnage(self, angle):
        rocket = armageddon.Armageddon()
        if angle > 0:
            rocket.RIGHT(1000)
        else:
            rocket.LEFT(1000)

    def camCallback(self, ballLandInfo):
        print 'cam callback'
        x = ballLandInfo.x
        y = ballLandInfo.y
        t = ballLandInfo.type
        d = ballLandInfo.distance
        angle = x/300.0*22.5
        angle = angle * -1.0
        rocket = rocket_msg()
        rocket.angle = angle
        if angle < 10 and angle > -10:
            rocket.fire = 1
        else:
            rocket.fire = 0
        rocket.stop = 0
        self.rocketPub.publish(rocket)

    def main_loop(self):
        self.init_publishers()
        #print "loop"
        self.init_subcribers()
        #self.init_params
        rate = rospy.Rate(10);

        while not rospy.is_shutdown():
            #put code here
            #print "while loop"
            rate.sleep()

if __name__ == '__main__':
    try:
        print "main"
        rocket = rocket_node()
        print "rocket node"
        rocket.main_loop()



    except rospy.ROSInterruptException:
        pass
